src/main/java/net/md_5/specialsource/Jar.java

```diff
@@ -75,8 +75,17 @@
      * @return
      */
     public boolean containsClass(String clazz) {
+        if (contains.contains(clazz)) {
+            return true;
+        }
+
         try {
-            return contains.contains(clazz) ? true : getClass(clazz) != null;
+            InputStream current = getClass(clazz);
```

Comment:
This was the minimal change to fix this.

Possibly it would be more correct to add a method getZipEntry which is like getResource but doesn't try to open the entry?

However in that case if trying to open the entry would fail then we could incorrectly return true in containsClass. Not sure if we ever actually would get a non-null result from getEntry but fail to open the stream in practice.